contrib/natsio/src/main/java/com/google/cloud/dataflow/contrib/natsio/NatsIO.java

```diff
@@ -0,0 +1,359 @@
+ *			.apply(ParDo.of(new NatsIOBench.SimpleFunc()));
+ * } </pre>
+ * 
+ * <p>or
+ * <pre> {@code
+ * 		options.setStreaming(true);
+ * 
+ * 		Pipeline p = Pipeline.create(options);
+ *		p
+ *			.apply(NatsIO.Read.from(subject, props)) // Streaming
+ *			.apply(ParDo.of(new NatsIOBench.SimpleFunc()));
+ * } </pre>
+ * 
+ */
+public class NatsIO {
+	private static final Logger LOG = LoggerFactory.getLogger(NatsIO.class);
+	
+	private static final Coder<String> DEFAULT_CODER = StringUtf8Coder.of();
+	private static CommitMark DEFAULT_MARK = new CommitMark();
+	private static final byte[] DEFAULT_ID = new byte[1];
+
+	public static class Read {
+		/**
+		 * Create a unbounded NatsSource instance. 
+		 * @param subject to subscribe and receive updates
+		 * @param props contains NATS related properties. "servers" property is mandatory to access a remote NATS server.
+		 * @return a unbounded NatsSource instance
+		 */
+		public static com.google.cloud.dataflow.sdk.io.Read.Unbounded<KV<String,String>> from(String subject, Properties props) {
+			return com.google.cloud.dataflow.sdk.io.Read.from(new NatsSource(subject, props));
+		}
+
+		/**
+		 * Create a NatsSource instance which is bounded to a number of updates to receive. 
+		 * @param subject to subscribe and received updates
+		 * @param numRecords bounds a number of records to receive from NATS server
+		 * @param props contains NATS related properties. "servers" property is mandatory to access a remote NATS server.
+		 * @return a bounded NatsSource instance
+		 */
+		public static PTransform<PInput, PCollection<KV<String,String>>> withMaxNumRecords(String subject, long numRecords, Properties props) {
+			return from(subject, props).withMaxNumRecords(numRecords);
+		}		
+
+		/**
+		 * Create a NatsSource instance which is bounded to total reading time to receive updates.
+		 * @param subject to subscribe and received updates
+		 * @param readTime bounds total reading time to receive updates from NATS server.
+		 * @param props contains NATS related properties. "servers" property is mandatory to access a remote NATS server.
+		 * @return a bounded NatsSource instance
+		 */
+		public static PTransform<PInput, PCollection<KV<String,String>>> withMaxReadtime(String subject, long readTime, Properties props) {
+			return from(subject, props).withMaxReadTime(Duration.standardSeconds(readTime));
+		}		
+		
+		public static class NatsSource extends UnboundedSource<KV<String,String>, UnboundedSource.CheckpointMark> {
+			private static final long serialVersionUID = 1L;
+			
+			private Properties props = null;
+			private Connection conn = null;
+			private String subject;
+			
+			public NatsSource(String subject, Properties props) {
+				this.subject = subject;
+				this.props = props;
+			}
+			
+			@Override
+			public NatsReader createReader(PipelineOptions arg0, UnboundedSource.CheckpointMark arg1) {
```

Comment:
in general, would prefer better variable names throughout than `arg0`, `arg1`. Any reason not to use the original names from the overridden function?